docs/source/scilla-in-depth.rst

```diff
@@ -1112,7 +1133,63 @@ which can be used to traverse all the Peano numbers from a given
   (of type ``'A``) when all Peano numbers down to ``Zero`` have been
   processed.
 
+- ``nat_foldk: ('A -> Nat -> ('A -> 'A) -> 'A) -> 'A -> Nat -> 'A``:
+  See below for an example and explanation.
```

Comment:
Can we please have a high-level description here? Like the one we have for `nat_fold` and the list folds.